webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { ComponentFixture, fakeAsync, flush, TestBed, tick } from '@angular/core/testing';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { TranslateFakeLoader, TranslateLoader, TranslateModule } from '@ngx-translate/core';
+import { expect } from 'chai';
+import * as moment from 'moment';
+import sinon from 'sinon';
+
+import { ChangesService } from '@mm-services/changes.service';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { RulesEngineService } from '@mm-services/rules-engine.service';
+import { TasksActions } from '@mm-actions/tasks';
+import { TelemetryService } from '@mm-services/telemetry.service';
+import { TourService } from '@mm-services/tour.service';
+import { TasksComponent } from '@mm-modules/tasks/tasks.component';
+import { NavigationComponent } from '@mm-components/navigation/navigation.component';
+import { Selectors } from '@mm-selectors/index';
+
+describe('TasksComponent', () => {
+  let getComponent;
+  let Changes;
+  let RulesEngine;
+  let Telemetry;
+  let Tour;
+  let contactTypesService;
+  let clock;
+  let store;
+
+  let component: TasksComponent;
+  let fixture: ComponentFixture<TasksComponent>;
+
+  beforeEach(async () => {
+    Changes = sinon.stub();
+    RulesEngine = {
+      isEnabled: sinon.stub().resolves(true),
+      fetchTaskDocsForAllContacts: sinon.stub().resolves([]),
+    };
+
+    Tour = sinon.stub();
+    Telemetry = { record: sinon.stub() };
+    contactTypesService = {
+      includes: sinon.stub(),
+    };
+
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: TranslateFakeLoader } }),
+        RouterTestingModule,
+      ],
+      providers: [
+        provideMockStore(),
+        { provide: ChangesService, useValue: { subscribe: Changes }},
+        { provide: RulesEngineService, useValue: RulesEngine },
+        { provide: TelemetryService, useValue: Telemetry },
+        { provide: TourService, useValue: { startIfNeeded: TourService } },
+        { provide: ContactTypesService, useValue: contactTypesService }
+      ],
+      declarations: [
+        TasksComponent,
+        NavigationComponent,
+      ],
+    });
+
+    getComponent = () => {
+      return TestBed.compileComponents().then(() => {
+        fixture = TestBed.createComponent(TasksComponent);
+        component = fixture.componentInstance;
+        store = TestBed.inject(MockStore);
+        fixture.detectChanges();
+      });
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    clock?.restore();
+  });
+
+  it('should ngOnDestroy should unsubscribe and clear state', async () => {
+    await getComponent();
+
+    const setTasksList = sinon.stub(TasksActions.prototype, 'setTasksList');
```

Comment:
Optional: I see unit test with 3 minimum sections and usually seperate them with some spaces in between for adding to the readability:
```
it('should..., () => {
  // Sec 1: Preparing for testing mocks, stubs, etc

 // Sec 2: Call the unit to test

// Sec 3: asserts, expects 
});
```
Some times the unit test structure accommodates to the need, like repeating Sec 2 and Sec 3: 
```
it('should..., () => {
  // Sec 1: Preparing for testing mocks, stubs, etc

 // Sec 2: Call the unit to test
 // Sec 3: asserts, expects

 // Sec 2: Call the unit to test
 // Sec 3: asserts, expects 
});
```

Reply:
I tried to separate assertions more, based on "concern"